### PR TITLE
Feature/webdatafiledumps

### PR DESCRIPTION
**Pull request by vinay-ebi**
The purpose of this pipeline is to automate current manual processing of the 7 species (https://2020.ensembl.org/app/species-selector) currently available on 2020 website.

### CODE REVIEW

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+=pod
+
+=head1 NAME
+
+Bio::EnsEMBL::Production::Pipeline::PipeConfig::SearchDumps_conf
```
Comment (marcoooo) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:24]: Does not correspond to the package name.

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+=head1 DESCRIPTION
+
+Pipeline to generate the Solr search, EBeye search and Advanced search indexes
```
Comment (marcoooo) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:28]: Please review documentation to be explicit about purpose / required parameters, may be an example of init command...

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                           'text'    => 'Output location : '. $self->o('output_path'), 
+                          },
+        },
+
+ 
+
+    ];
+} ## end sub pipeline_analyses
+
+sub beekeeper_extra_cmdline_options {
```
Comment (marcoooo) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:145]: Don't think this is needed anymore. see comment from @james-monkeyshines  here: https://github.com/Ensembl/ensembl-production/pull/452/files#r456581282

File: modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+ Bio::EnsEMBL::Production::Pipeline::Webdatafile::WebSpeciesFactory;
```
Comment (marcoooo) [inline modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm:20]: Not corresponding to actual package name.

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 1 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm]: resolved

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 2 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm]: added new description

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 3 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm]: base package is added as per suggestion

File: modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm (hunk 4 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm]: resolved

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            -flow_into  => { '2' => [ 'StepBootstrap' ],
+                             #'A->1' => ['email_notification']
+                           },
+             
+        },
+        {    
+            -logic_name => 'StepBootstrap',
+            -module     => 'Bio::EnsEMBL::Production::Pipeline::Webdatafile::WebdataFile',
+            -parameters => { current_step       => 'bootstrap',
+                             step => $self->o('step',)
```
Comment (thomasmaurel) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:122]: Typo here, "," should be outside the o('')

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+=head1 NAME
+
+Bio::EnsEMBL::Production::Pipeline::PipeConfig::WebDataFile_conf
+
+=head1 DESCRIPTION
+
+This pipeline automate current manual processing of the 7 species (https://2020.ensembl.org/app/species-selector) currently available on 2020 website.
+
+registry=${BASR_DIR}/ensembl-production_private/release_coordination/registries/reg_ens-sta-1_rw.pm
```
Comment (thomasmaurel) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:30]: Not sure we need this in the module. Usually we don't publish the SOP.

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+          -logic_name  => 'GeneInfo_logic',
+          -input_ids  => [ {} ],
+          -module      =>  'Bio::EnsEMBL::Production::Pipeline::Webdatafile::GenomeInfoYml',
+          -parameters => { species => $self->o('species'),
+                           genomeinfo_yml => $self->o('genomeinfo_yml'), 
+                           run_all        => $self->o('run_all'),   
+                         },
+          -flow_into   =>  {  '2->A' => ['StepBootstrap'],
+                              '3->A'  => ['SpeciesFactory'],   
+                               # WHEN(
```
Comment (thomasmaurel) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:94]: Maybe this comment can be cleaned up now?

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+                              'A->1' => ['email_notification'],   
+                           }
+
+        }, 
+
+        {
+            -logic_name => 'SpeciesFactory',
+            -module     =>
+                'Bio::EnsEMBL::Production::Pipeline::Common::SpeciesFactory',
+            #-input_ids  => [ {} ], # required for automatic seeding
```
Comment (thomasmaurel) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:107]: Maybe this comment can be cleaned up now?

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+            -parameters => { species => $self->o('species'),
+                antispecies          => $self->o('antispecies'),
+                division             => $self->o('division'),
+                run_all              => $self->o('run_all') 
+             },
+            -flow_into  => { '2' => [ 'StepBootstrap' ],
+                             #'A->1' => ['email_notification']
```
Comment (thomasmaurel) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:114]: Maybe this comment can be cleaned up now?

File: modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+sub run {
+
+  my ($self) = @_;
+  my $species = $self->param('species');
+  my $current_step = $self->param('current_step') ;
+  my $output = $self->param('output_path');
+  my $app_path = $self->param('app_path'); 
+  my $log_file =  "$output/WebFiles_$current_step.log";
+  my $cmd = "$app_path/bin/prepare_all_datafiles.pl $species -output $output -step $current_step &> $log_file";
+  #my ($rc, $sysoutput)=$self->run_cmd($cmd);
```
Comment (thomasmaurel) [inline modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm:56]: Maybe this comment can be cleaned up now?

File: modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+  if ($current_step eq 'bootstrap') {
+        my @steps     = @{$self->param('step')};
+        my %each_flow = (
+             'transcripts' => 1,
+             'contigs'     => 2,
+             'gc'          => 3,
+             'variants'    => 4,
+        );
+  
+        #my @all_steps = (['transcripts', 1], ['contigs', 2], ['gc', 3], ['variants', 4]);
```
Comment (thomasmaurel) [inline modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm:86]: Maybe this comment can be cleaned up now?

File: modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+        my @existing_steps = keys %each_flow;
+        my @all_steps = scalar intersect(@steps, @existing_steps) ? intersect(@steps, @existing_steps) : @existing_steps;
+        for my $each_step (@all_steps){
+            #$self->warning($each_step);
```
Comment (thomasmaurel) [inline modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm:90]: Maybe this comment can be cleaned up now?

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 6 above)
Comment (marcoooo) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:30]: I asked for it :-) sorry @vinay-ebi .

Comment (marcoooo) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:30]: we then need a confluence SOP page

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 5 above)
Comment (marcoooo) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:122]:
```suggestion
                             step => $self->o('step'),
```

Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm]: resolved

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 6 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:30]: sure i will create a SOP in confluence page

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 7 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm]: resolved

File: modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm (hunk 10 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFile.pm]: resolved

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+This pipeline automate current manual processing of the 7 species (https://2020.ensembl.org/app/species-selector) currently available on 2020 website.
+
+
+=cut
+
+package Bio::EnsEMBL::Production::Pipeline::PipeConfig::WebDataFile_conf;
+use strict;
+use Bio::EnsEMBL::Hive::PipeConfig::HiveGeneric_conf;
+use base ('Bio::EnsEMBL::Production::Pipeline::PipeConfig::Base_conf');
+use Bio::EnsEMBL::ApiVersion qw/software_version/;
```
Comment (james-monkeyshines) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:37]: Not needed.

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+
+use Data::Dumper;
```
Comment (james-monkeyshines) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:39]: Not used.

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+        %{$self->SUPER::default_options()},
+        species         => [],
+        division        => [],
+        antispecies     => [],
+        run_all         => 0, #always run every species
+        step             => [],  
+        genomeinfo_yml    => undef,
+        # Email Report subject
+        email_subject => $self->o('pipeline_name').'  pipeline has finished',
+        release => software_version()
```
Comment (james-monkeyshines) [inline modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:53]: Neither of these parameters is used.

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 13 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:37]: i will remove it

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 14 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:39]: used for debug forgot to remove it

File: modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm (hunk 15 above)
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:53]: i will add email_subject  in  analysis 'email_notification' as subject parameter  
release parameter currently not used but will be used in future

Comment (james-monkeyshines) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:53]: The "release" variable is available via the parent conf, but it's called `ensembl_release`.

Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/WebDataFile_conf.pm:53]: sure i will make use of this variable

**Pull request merged**